mantis-server/mantis-server-worker/src/main/java/io/mantisrx/server/worker/TaskExecutor.java

```diff
@@ -126,9 +128,22 @@ public TaskExecutor(
         MachineDefinition machineDefinition =
             MachineDefinitionUtils.sys(workerPorts, workerConfiguration.getNetworkBandwidthInMB());
         String hostName = workerConfiguration.getExternalAddress();
+        String containerDefinitionId = System.getenv(WorkerConstants.WORKER_CONTAINER_DEFINITION_ID);
```

Comment:
Instead of defining a specific attribute and looking for it, why not pass all attributes as part of some env (in JSON or another format, e.g., `TASK_EXECUTOR_ATTRIBUTES`) and then decode that and send it as part of registration? This would not leak your implementation details, I think.